pkg/agent/client.go

```diff
@@ -439,6 +450,10 @@ func (a *Client) Serve() {
 				dialResp.GetDialResponse().ConnectID = connID
 				if err := a.Send(dialResp); err != nil {
 					klog.ErrorS(err, "could not send dialResp")
+					// clean-up is normally called from remoteToProxy which we will never invoke.
+					// So we are invoking it here to force the clean-up to occur.
+					closeOnce.Do(func() { close(dialDone) })
+					connCtx.cleanup()
```

Comment:
Another way we could handle this is to just call `go connCtx.cleanup()` (and go back to the regular deferred close) in a go-routine. I'm not sure that's better, I just find the `closeOnce.Do(func() { close(dialDone) })` aversive, but that might just be aesthetics...

Reply:
Made the switch but also added a comment explaining why we have the extra goroutine as  don't think its obvious on first read through.